### PR TITLE
Export resume function

### PR DESCRIPTION
**Pull request by cpfiffer**
`resume` was not being exported, so I fixed that and made it more intuitive by automatically concatenating samples to the old chain. Tests won't pass until MCMCChains has a new minor release.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -36,8 +36,8 @@ DistributionsAD = "0.1.0"
 FiniteDifferences = "0.9"
 ForwardDiff = "0.10.3"
 Libtask = "0.3.1"
+MCMCChains = "^0.3.15"
```
Comment (devmotion) [inline Project.toml]: The [caret specifiers are the default version specifiers](https://julialang.github.io/Pkg.jl/v1/compatibility/#Version-specifier-format-1), so it's equivalent to writing just "0.3.15" without caret. Maybe you actually wanted to use a tilde specifier?

Comment (cpfiffer) [inline Project.toml]:
```suggestion
MCMCChains = "0.3.15"
```

Nope, I didn't mean to use a tilde, but I will remove the caret.

Comment (cpfiffer) [reply Project.toml]: Thanks for catching that, though, I wasn't aware that it was the default version spec.

**Comment by cpfiffer** [comment]
Tests are passing on this one, could someone take a look at this?

**Pull request merged**